prefect_azure/blob_storage.py

```diff
@@ -177,3 +191,509 @@ def example_blob_storage_list_flow():
+        )
+        full_container_path = self._get_path_relative_to_base_folder(to_path)
+        async with self.credentials.get_blob_client(
+            self.container_name, full_container_path
+        ) as blob_client:
+            try:
+                await blob_client.upload_blob(from_file_object, **upload_kwargs)
+            except ResourceNotFoundError as exc:
+                raise RuntimeError(
+                    f"An error occurred when atempting to download from container {self.container_name}: {exc.reason}"
+                ) from exc
+
+        return to_path
+
+    @sync_compatible
+    async def upload_from_path(
+        self, from_path: Union[str, Path], to_path: str, **upload_kwargs: Dict[str, Any]
+    ) -> Coroutine[Any, Any, str]:
+        """
+        Uploads an object from a local path to the specified destination path in the
+            blob storage container.
+
+        Args:
+            from_path: The local path of the object to upload.
+            to_path: The destination path in the blob storage container.
+            **upload_kwargs: Additional keyword arguments to pass to the
+                `upload_blob` method.
+
+        Returns:
+            The destination path in the blob storage container.
+
+        Example:
+            Upload a file from the local path `file.txt` to the container
+                at the path `object`:
+
+            ```python
+            from prefect_azure import AzureBlobStorageCredentials
+            from prefect_azure.blob_storage import AzureBlobStorageContainer
+
+            credentials = AzureBlobStorageCredentials(
+                connection_string="connection_string",
+            )
+            block = AzureBlobStorageContainer(
+                container_name="container",
+                credentials=credentials,
+            )
+            block.upload_from_path(
+                from_path="file.txt",
+                to_path="object"
+            )
+            ```
+        """
+        self.logger.info(
+            "Uploading object to container %s with key %s", self.container_name, to_path
+        )
+        full_container_path = self._get_path_relative_to_base_folder(to_path)
+        async with self.credentials.get_blob_client(
+            self.container_name, full_container_path
+        ) as blob_client:
+            try:
+                with open(from_path, "rb") as f:
+                    await blob_client.upload_blob(f, **upload_kwargs)
+            except ResourceNotFoundError as exc:
+                raise RuntimeError(
+                    f"An error occurred when atempting to download from container {self.container_name}: {exc.reason}"
+                ) from exc
+
+        return to_path
+
+    @sync_compatible
+    async def upload_from_folder(
+        self,
+        from_folder: Union[str, Path],
+        to_folder: str,
+        **upload_kwargs: Dict[str, Any],
+    ) -> Coroutine[Any, Any, str]:
```

Comment:
```python
credentials = AzureBlobStorageCredentials(
    connection_string=my_connection_string,
)
container = AzureBlobStorageContainer(
    container_name=existing_container,
    credentials=credentials,
)
container.upload_from_folder("nonexistentlocalfolder/", to_folder="test_folder")
```

This seems to silently fail if I have no local folder with that name. No `test_folder` gets created.

Reply:
Added a check in https://github.com/PrefectHQ/prefect-azure/pull/139/commits/d0b195245228dc4bab20663c343cd0dfd8134136

Reply:
nice!
```bash
❯ python qa.py
13:31:28.118 | INFO    | prefect.AzureBlobStorageContainer - Uploading folder to container testing314 with key docsx
Traceback (most recent call last):
  File "/Users/bean/code-oss/prefect-azure/qa.py", line 15, in <module>
    container.upload_from_folder(from_folder="docfadfdss/", to_folder="docsx")
  File "/Users/bean/code-oss/prefect-azure/azurevenv/lib/python3.10/site-packages/prefect/utilities/asyncutils.py", line 259, in coroutine_wrapper
    return call()
  File "/Users/bean/code-oss/prefect-azure/azurevenv/lib/python3.10/site-packages/prefect/_internal/concurrency/calls.py", line 431, in __call__
    return self.result()
  File "/Users/bean/code-oss/prefect-azure/azurevenv/lib/python3.10/site-packages/prefect/_internal/concurrency/calls.py", line 317, in result
    return self.future.result(timeout=timeout)
  File "/Users/bean/code-oss/prefect-azure/azurevenv/lib/python3.10/site-packages/prefect/_internal/concurrency/calls.py", line 178, in result
    return self.__get_result()
  File "/Users/bean/.pyenv/versions/3.10.4/lib/python3.10/concurrent/futures/_base.py", line 391, in __get_result
    raise self._exception
  File "/Users/bean/code-oss/prefect-azure/azurevenv/lib/python3.10/site-packages/prefect/_internal/concurrency/calls.py", line 388, in _run_async
    result = await coro
  File "/Users/bean/code-oss/prefect-azure/prefect_azure/blob_storage.py", line 598, in upload_from_folder
    raise ValueError(f"{from_folder} is not a directory")
ValueError: docfadfdss/ is not a directory
```